stdlib/LinearAlgebra/test/diagonal.jl

```diff
@@ -238,6 +238,10 @@ Random.seed!(1)
         @test D2*transpose(D) ≈ Array(D2)*transpose(Array(D))
         @test D2*D' ≈ Array(D2)*Array(D)'
 
+        # type-stable division for pseudo-diagonal matrices
+        @test (@inferred Matrix(D) \ Matrix(D2)) ≈ D \ D2
```

Comment:
The inference test involving matrices appears to pass without this change as well?
```julia
julia> @inferred Matrix(Diagonal(rand(4))) \ Matrix(Diagonal(rand(4)));
```

An inference test that appears to fail without this change is
```julia
julia> @inferred rand(4,4) \ Bidiagonal(fill(3,4), fill(3,3), :U)
ERROR: return type Matrix{Float64} does not match inferred return type Union{Bidiagonal{Float64, Vector{Float64}}, LowerTriangular{Float64, Matrix{Float64}}, UpperTriangular{Float64, Matrix{Float64}}, Matrix{Float64}}
```

Reply:
```suggestion
        @test (@inferred Matrix(D) \D2) ≈ D \ D2
```

Reply:
Indeed, I lost the original issue out of sight.